packages/toolkit/src/createAsyncThunk.ts

```diff
@@ -553,11 +553,15 @@ If you want to use the AbortController to react to \`abort\` events, please cons
       const promise = (async function () {
         let finalAction: ReturnType<typeof fulfilled | typeof rejected>
         try {
-          if (
-            options &&
-            options.condition &&
-            options.condition(arg, { getState, extra }) === false
-          ) {
+          let conditionResult = options?.condition?.(arg, { getState, extra })
+          if (conditionResult instanceof Promise) {
```

Comment:
Is `instanceof` good enough or do we want to support 3rd-party thenables?

Reply:
Also what should happen when an error is thrown in a synchronous `condition`?

Reply:
Hmm. `"then" in conditionResult" might be better, you're right.

As for an error in `condition`, so far that was just thrown as it was and I'm pretty okay. Might even make sense to remove the try..catch block above to mimick that.

Reply:
> As for an error in condition, so far that was just thrown as it was and I'm pretty okay. Might even make sense to remove the try..catch block above to mimick that.

Okay, let's do it that way. Bur I somewhat dislike the fact that in this case `rejected` is dispatched without corresponding `pending`.